analysis/cpp/src/make-histograms-loop.cpp

```diff
@@ -115,22 +140,45 @@ int main(int argc, char* argv[]) {
         // h_vtzzz->Fill(pv_zpos * 1e3);
 
         // loop over the MC particles
-        // for(size_t iparticle = 0; iparticle < n_mc_particles_in_event; ++iparticle) {
-        //     int pdgid = mcpdg[iparticle];
-
-        //     // skip things we don't care about
-        //     if(std::abs(pdgid) > 2000) continue;
-
-        //     h_mcpdg->Fill(pdgid);
-
-        //     // MC particle momentum of this particle
-        //     double px = mc_px[iparticle];
-        //     double py = mc_px[iparticle];
-        //     double transverse_momentum = sqrt(px*px + py*py); // GeV
-        //     if(transverse_momentum > 1.0) {
-        //         h_mcpdg_ptcut->Fill(pdgid);
-        //     }
-        // } // loop over iparticle
+        float mu_vtx_x_pos = 0;
+        float antimu_vtx_x_pos = 0;
+        unsigned n_muons_in_event = 0;
+        for(size_t iparticle = 0; iparticle < n_mc_particles_in_event; ++iparticle) {
+            int pdgid = mcpdg[iparticle];
+            int generator_status = mcgst[iparticle];
+
+            // skip things we don't care about
+            if(std::abs(pdgid) > 2000) continue;
+
+            // let's only grab the muons with generator status == 1
+            if(std::abs(pdgid) == 13 && generator_status == 1) {
+                n_muons_in_event++;
+                // for the MC vertex that we want to compare to the reco level PV, let's use the muons (not antimuons)
+                if(pdgid == 13) {
+                    mu_vtx_x_pos = mc_vtx_xpos[iparticle] * 1e6;
+                    h_mcvtx_muon->Fill(mu_vtx_x_pos);
+                    h_mc_muon_count->Fill(1.5);
+                } else if(pdgid == -13) {
+                    antimu_vtx_x_pos = mc_vtx_xpos[iparticle] * 1e6;
+                    h_mc_muon_count->Fill(-1.5);
+                }
+            }
+
+            //h_mcpdg->Fill(pdgid);
+
+            //// MC particle momentum of this particle
+            //double px = mc_px[iparticle];
+            //double py = mc_px[iparticle];
+            //double transverse_momentum = sqrt(px*px + py*py); // GeV
+            //if(transverse_momentum > 1.0) {
+            //    h_mcpdg_ptcut->Fill(pdgid);
+            //}
+        } // loop over iparticle
+
+        //h_mc_muon_count->Fill(n_muons_in_event);
+        float delta_mu_vtx_x = mu_vtx_x_pos - antimu_vtx_x_pos;
+        h_mu_mcvtx_diff->Fill(delta_mu_vtx_x);
```

Comment:
Compute the difference between the generator status `==1` muon and antimuon MC vertex x-positions and fill the histogram.